runtime/mlBytes.js

```diff
@@ -89,8 +89,7 @@ function caml_utf8_of_utf16(s) {
     if (c < 0x80) {
       for (var j = i + 1; j < l && (c = s.charCodeAt(j)) < 0x80; j++);
       if (j - i > 512) {
-        t.substr(0, 1);
-        b += t;
+        b += t.slice(0, 1);
```

Comment:
This is just wrong. other changes about `substr` are wrong as well.

Reply:
Oops... Since neither `substr` nor `slice` manipulate the original array, the operation itself was actually unnecessary.

Reply:
It was here on purpose, to force the js engine to fatten the string and improve performance. This is yet another example of things that are not as obvious as they seems. Can you leave this to another PR, we might need to benchmark

Reply:
Done